src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,150 @@
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    @State()
+    private invalidLink: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
+
+    public render() {
+        return [
+            <limel-input-field
+                label="Text"
+                value={this.link?.text || ''}
+                leadingIcon="text_cursor"
+                onChange={this.handleLinkTitleChange}
+            />,
+            <limel-input-field
+                label="Link"
+                value={this.link?.href || ''}
+                leadingIcon="-lime-text-link"
+                invalid={this.invalidLink}
+                onChange={this.handleLinkValueChange}
+            />,
+            <div class="actions">
+                <limel-button
+                    label={this.getTranslation('cancel')}
+                    onClick={this.handleCancel}
+                    slot="button"
+                />
+                <limel-button
+                    primary={true}
+                    label={this.getTranslation('save')}
+                    disabled={!this.link?.href || this.invalidLink}
+                    onClick={this.handleSave}
+                    slot="button"
+                />
+            </div>,
+        ];
+    }
+
+    private getTranslation(key: string) {
+        return translate.get(`${key}`, this.language);
+    }
+
+    private validateLink = (href: string): boolean => {
+        if (this.isValidLink(href)) {
+            return true;
+        }
+
+        // If the link doesn't have a scheme, add 'http://' and validate again
+        if (!/^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(href)) {
+            const formattedLink = 'http://' + href;
+            if (this.isValidLink(formattedLink)) {
+                this.link = { ...this.link, href: formattedLink };
+
+                return true;
+            }
+        }
+
+        return false;
+    };
+
+    private isValidLink = (href: string): boolean => {
+        try {
+            const url = new URL(href);
+
+            return url.protocol === 'http:' || url.protocol === 'https:';
+        } catch {
+            return false;
+        }
+    };
```

Comment:
We are doing almost the same thing here:

<img width="747" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/154725/6f5ae750-368c-459a-a35e-ed9b365cf38e">

We also have a whole heap of similar stuff in https://github.com/Lundalogik/lime-elements/blob/main/src/util/link-helper.ts

I don't think we will want to limit the possible protocols to `http` and `https`. We've limited the possible protocols in https://github.com/Lundalogik/lime-elements/blob/main/src/util/link-helper.ts for some reason that I can't quite remember right now, and it's already become a problem because there are all kinds of app-specific protocols people want to be able to use.

So, we should reuse what's already in https://github.com/Lundalogik/lime-elements/blob/main/src/util/link-helper.ts, and adapt it if needed. The functions in there that aren't already exported can certain be exported so they are possible to import and use here and in `menu-commands.ts`.

Reply:
You are absolutely right 👍.
My intention was to consolidate that before putting it in review. There are more things like this since I had to hand over this change mid-process 🙈

Reply:
I've gone ahead and used the functions in `link-helper.ts` for validation.